### PR TITLE
settingsChats fragment

### PR DESCRIPTION
**Pull request by UgurDenli**

### CODE REVIEW

File: WhatsApp/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -32,4 +32,26 @@
...
+    <string name="mediaVisibilityDesc">Show newly downloaded media\nin your phone\'s gallery</string>
+    <string name="fontSize">Font Size</string>
+    <string name="medium">Medium</string>
+    <string name="chatBackup">Chat Backup</string>
+    <string name="appLanguage">App Language</string>
+    <string name="phoneLanguage">Phone\'s language</string>
+    <string name="chatHistory">Chat History</string>
+    <string name="archivedChats">Archived chats</string>
+    <string name="archivedChatsTitle">Keep chats archived</string>
+    <string name="archivedChatsDesc">Archived chats will remain\narchived when you receive a\nnew message</string>
```
Comment (OskarLasota) [inline WhatsApp/app/src/main/res/values/strings.xml]: this doesnt need a /n tag, it just needs margin

File: WhatsApp/app/src/main/res/layout/fragment_settings_chat.xml [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+        <View
+            android:id="@+id/view"
+            android:layout_width="match_parent"
+            android:layout_height="1dp"
+            android:layout_marginVertical="16dp"
+            android:background="@color/grey"
+            app:layout_constraintTop_toBottomOf="@+id/wallpaperTransfer" />
+
+        <RelativeLayout
+            android:id="@+id/enterIsSend"
```
Comment (OskarLasota) [inline WhatsApp/app/src/main/res/layout/fragment_settings_chat.xml:109]: this needs a subheading, like on original whatsap design

File: WhatsApp/app/src/main/res/layout/fragment_settings_chat.xml [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+    android:id="@+id/root_layout"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
+
+    <LinearLayout
+        android:orientation="vertical"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+
+    <RelativeLayout
```
Comment (OskarLasota) [inline WhatsApp/app/src/main/res/layout/fragment_settings_chat.xml]: i dont think we need a layout wrapper for this

File: WhatsApp/app/src/main/res/layout/fragment_settings_chat.xml [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="16dp"
+            android:text="@string/display" />
+
+    </RelativeLayout>
+
+        <RelativeLayout
+            android:id="@+id/themeTransfer"
```
Comment (OskarLasota) [inline WhatsApp/app/src/main/res/layout/fragment_settings_chat.xml:29]: the text margin between the left side of the screen should be the same for all the rows.

File: WhatsApp/app/src/main/res/layout/fragment_settings_chat.xml (hunk 3 above)
Comment (UgurDenli) [reply WhatsApp/app/src/main/res/layout/fragment_settings_chat.xml]: its used for spacing reasons

File: WhatsApp/app/src/main/res/layout/fragment_settings_chat.xml [hunk 5]
```diff
@@ -267,8 +270,9 @@
                 android:layout_height="wrap_content"
                 android:layout_below="@+id/tvArchivedChats"
                 android:layout_alignParentStart="true"
-                android:layout_toEndOf="@id/archivedSwitch"
                 android:layout_marginStart="70dp"
+                android:layout_marginEnd="140dp"
+                android:layout_toEndOf="@id/archivedSwitch"
                 android:text="@string/archivedChatsDesc" />
         </RelativeLayout>
```
Comment (OskarLasota) [inline WhatsApp/app/src/main/res/layout/fragment_settings_chat.xml:274]: for this relative layout, here is an example of how to use the alignment properties. Please do the same for every row and keep the marginEnd to max 20dp.

`        <RelativeLayout
            android:id="@+id/archivedChatsLayout"
            android:layout_width="match_parent"
            android:layout_height="wrap_content"
            android:clickable="true"
            android:focusable="true"
            android:paddingTop="12dp"
            android:paddingBottom="12dp"
            android:background="?attr/selectableItemBackground">

            <com.google.android.material.switchmaterial.SwitchMaterial
                android:id="@+id/archivedSwitch"
                android:layout_width="wrap_content"
                android:layout_height="wrap_content"
                android:layout_alignParentRight="true"
                android:checked="false" />


            <TextView
                android:id="@+id/tvArchivedChats"
                style="@style/chatSettingsText"
                android:layout_width="wrap_content"
                android:layout_height="wrap_content"
                android:layout_alignParentStart="true"
                android:layout_marginStart="70dp"
                android:layout_toStartOf="@id/archivedSwitch"
                android:text="@string/archivedChatsTitle" />

            <TextView
                android:id="@+id/tvArchivedChatsDesc"
                android:layout_width="wrap_content"
                android:layout_height="wrap_content"
                android:layout_below="@+id/tvArchivedChats"
                android:layout_alignParentStart="true"
                android:layout_marginStart="70dp"
                android:layout_marginEnd="20dp"
                android:layout_toStartOf="@id/archivedSwitch"
                android:text="@string/archivedChatsDesc" />
        </RelativeLayout>`

File: WhatsApp/app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -51,7 +51,7 @@
     <string name="chatHistory">Chat History</string>
     <string name="archivedChats">Archived chats</string>
     <string name="archivedChatsTitle">Keep chats archived</string>
-    <string name="archivedChatsDesc">Archived chats will remain\narchived when you receive a\nnew message</string>
+    <string name="archivedChatsDesc">Archived chats will remain archived when you receive a new message</string>
```
Comment (OskarLasota) [inline WhatsApp/app/src/main/res/values/strings.xml:84]: better!

File: WhatsApp/app/src/main/res/values/styles.xml [hunk 7]
```diff
@@ -10,6 +10,11 @@
         <item name="android:textColor">@color/black</item>
     </style>
 
+    <style name="chatSettingsText">
+        <item name="android:textSize">16sp</item>
```
Comment (NelzkieLabs) [inline WhatsApp/app/src/main/res/values/styles.xml:14]: we can use dimens xml for 16sp

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/chats/SettingsChatsFragment.kt [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        super.onCreate(savedInstanceState)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        observeUiInteractions()
+
+    }
+
+    private fun observeUiInteractions() {
```
Comment (Josh-Owen) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/chats/SettingsChatsFragment.kt:28]: You need to override initViews and add all of these listeners there

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/chats/SettingsChatsFragment.kt [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        setHasOptionsMenu(true)
+        super.onCreate(savedInstanceState)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        observeUiInteractions()
```
Comment (Josh-Owen) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/chats/SettingsChatsFragment.kt:24]: `onViewCreated` will not be needed once `observeUiInteractions ` has been removed

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/chats/SettingsChatsFragment.kt [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+@AndroidEntryPoint
+class SettingsChatsFragment : BaseFragment<FragmentSettingsChatBinding>() {
+
+
+    override fun inflateBinding(layoutInflater: LayoutInflater): FragmentSettingsChatBinding {
+        return FragmentSettingsChatBinding.inflate(layoutInflater)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        setHasOptionsMenu(true)
```
Comment (Josh-Owen) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/chats/SettingsChatsFragment.kt]: We wont need `onCreate` in this instance `setHasOptionsMenu` can be moved to `initViews` also

**Review by Josh-Owen — CHANGES_REQUESTED** [review]

**Review by Josh-Owen — DISMISSED** [review]
Looks good to me

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/chats/SettingsChatsFragment.kt (hunk 10 above)
Comment (Josh-Owen) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/chats/SettingsChatsFragment.kt]: I think we can remove this onCreate and add `setHasOptionsMenu(true)` to `initViews()`

**Review by OskarLasota — APPROVED** [review]

**Pull request merged**